### PR TITLE
ref: remove ifdef from SentryXamarinFormsIntegration

### PR DESCRIPTION
**Pull request by lucas-zimerman**
With that, the NativeEventProcessors are based on INativeEventProcessors (which is  also based on ISentryEventProcessor)

The main Addition to INativeEventProcessors is

```
        /// <summary>
        /// Informs if the Native Event Processor does any processing or not.
        /// </summary>
        bool Implemented { get; }

        /// <summary>
        /// The target name, Android, iOS, ...
        /// </summary>
        string TargetName { get; }
```

With that, SentryXamarinFOrmsIntegration is able to know if the Event Processor does anything and the target name for logging.

### CODE REVIEW

File: Sentry.Xamarin.Forms/Internals/INativeEventProcessor.cs [hunk 1]
```diff
@@ -0,0 +1,17 @@
+﻿using Sentry.Extensibility;
+
+namespace Sentry.Xamarin.Forms.Internals
+{
+    internal interface INativeEventProcessor : ISentryEventProcessor
+    {
+        /// <summary>
+        /// Informs if the Native Event Processor does any processing or not.
+        /// </summary>
+        bool Implemented { get; }
```
Comment (bruno-garcia) [inline Sentry.Xamarin.Forms/Internals/INativeEventProcessor.cs:10]: This looks super hacky and IMHO an indication there's a problem with the desing

File: Sentry.Xamarin.Forms/SentryXamarinFormsIntegration.cs [hunk 2]
```diff
@@ -43,9 +43,17 @@ public void Register(IHub hub, SentryOptions options)
             _hub = hub;
 
             options.AddEventProcessor(new XamarinFormsEventProcessor(options));
-#if !NETSTANDARD
-            options.AddEventProcessor(new NativeEventProcessor(options));
-#endif
+
+            var nativeEventProcessor = new NativeEventProcessor(options);
+            if (nativeEventProcessor.Implemented)
```
Comment (bruno-garcia) [inline Sentry.Xamarin.Forms/SentryXamarinFormsIntegration.cs:48]: yeah this shouldn't go like this.

Maybe you need a `#if NATIVE_PROCESSOR` and conditionally compile the whole thing

**Review by bruno-garcia — CHANGES_REQUESTED** [review]

File: Sentry.Xamarin.Forms/SentryXamarinFormsIntegration.cs (hunk 2 above)
Comment (lucas-zimerman) [reply Sentry.Xamarin.Forms/SentryXamarinFormsIntegration.cs:48]: if added

File: Sentry.Xamarin.Forms/SentryXamarinFormsIntegration.cs [hunk 3]
```diff
@@ -68,12 +74,12 @@ public void Register(IHub hub, SentryOptions options)
...
-                if (nativeIntegration.Implemented)
-                {
-                    nativeIntegration.Register(hub, options);
-                    Nativeintegration = nativeIntegration;
-                }
+#if NATIVE_INTEGRATION
+                Nativeintegration = new NativeIntegration(Options.Value);
+                Nativeintegration.Register(hub, options);
+#else
+                options.DiagnosticLogger.Log(SentryLevel.Info, "No NativeIntegration found for the given target.");
```
Comment (bruno-garcia) [inline Sentry.Xamarin.Forms/SentryXamarinFormsIntegration.cs:81]:
```suggestion
                options.DiagnosticLogger.Log(SentryLevel.Debug, "No NativeIntegration found for the given target.");
```

File: Sentry.Xamarin.Forms/Sentry.Xamarin.Forms.csproj [hunk 4]
```diff
@@ -15,7 +15,21 @@
 		<Description>Official Sentry SDK for Xamarin Forms - Open-source error tracking that helps developers monitor and fix crashes in real time.</Description>
 
 		<_WriteTelemetryProperties>false</_WriteTelemetryProperties>
+		<DefineConstants>$(AdditionalConstants)</DefineConstants>
 	</PropertyGroup>
+
+	<PropertyGroup Condition=" $(TargetFramework.StartsWith('Xamarinios')) ">
+		<DefineConstants>$(DefineConstants);NATIVE_PROCESSOR;NATIVE_INTEGRATION</DefineConstants>
```
Comment (bruno-garcia) [inline Sentry.Xamarin.Forms/Sentry.Xamarin.Forms.csproj:22]: Looks like you could merge both constants

**Review by bruno-garcia — APPROVED** [review]

**Pull request merged**